rosbag2_transport/test/rosbag2_transport/test_rosbag2_node.cpp

```diff
@@ -59,20 +60,28 @@ class RosBag2NodeFixture : public Test
   }
 
   std::vector<std::string> subscribe_raw_messages(
-    size_t expected_messages_number, const std::string & topic_name, const std::string & type)
+    size_t expected_messages_number,
+    const std::string & topic_name,
+    const std::string & type,
+    const rclcpp::QoS & qos = rosbag2_transport::Rosbag2QoS())
   {
+    using namespace std::chrono_literals;
+    auto start = std::chrono::system_clock::now();
     std::vector<std::string> messages;
     size_t counter = 0;
```

Comment:
Should `counter` be atomic here since the callback is incrementing it while you busy wait on the current thread?